src/api/global/interfaces/interfaces.wallet.js

```diff
@@ -96,3 +98,78 @@ export const useCheckWalletConnected = () => {
     getWalletConnectionProps,
   };
 };
+
+export const useWalletBalances = () => {
+  // Global Wallet State
+  const { state, dispatch } = useContext(storeContext);
+  const { provider, address, network } = state.wallet;
+  // User Selected Token
+  const { getBridgeBurnInputProps } = useBridgeBurnInput();
+  const { token } = getBridgeBurnInputProps();
+  // Balance States
+  const [balances, setBalances] = useState({
+    ETH: 0,
+    renBTC: 0,
+    WBTC: 0,
+    ibBTC: 0,
+    USDC: 0,
+  });
+
+  useEffect(() => {
+    // ETH TOKEN BALANCE
+    provider.getBalance(address).then((bal) => {
+      const balanceInEth = ethers.utils.formatEther(bal);
+      setBalances({
+        ...balances,
+        ETH: parseFloat(balanceInEth),
+      });
+    });
+  }, []);
```

Comment:
We should handle provider, address and chain changes here. When the user changes address we want to show the correct balances

Reply:
This was pretty simple to fix. All I had to do was add `token, address, network, provider` inside the dependents array for the useEffect. 

The reason why that makes it work is because those variables are all in the global context and they change when the user changes their wallet or network.